ChangeSlimeHutchLimit/ModEntry.cs

```diff
@@ -11,19 +16,134 @@ public class ModEntry : Mod
 
     public override void Entry(IModHelper helper)
     {
+        // Load the configuration
         _config = this.Helper.ReadConfig<Config>();
 
-        this.Helper.ConsoleCommands.Add("SetSlimeHutchLimit", "Changes the max number of slimes that can inhabit a slime hutch.\n\nUsage: SetSlimeHutchLimit <value>\n- value: the number of slimes", this.ChangeMaxSlimes);
+        // Add console commands
+        this.Helper.ConsoleCommands.Add("SetSlimeHutchLimit",
+            "Changes the max number of slimes that can inhabit a slime hutch.\n\nUsage: SetSlimeHutchLimit <value>\n- value: the number of slimes",
+            this.ChangeMaxSlimes);
 
+        // Set up Harmony patches
         Harmony harmony = new Harmony(this.ModManifest.UniqueID);
-        harmony.Patch(AccessTools.Method(typeof(SlimeHutch), nameof(SlimeHutch.isFull)), postfix: new HarmonyMethod(typeof(ModEntry), nameof(ModEntry.SlimeHutch_isFull_postfix)));
+
+        harmony.Patch(
+            AccessTools.Method(typeof(SlimeHutch), nameof(SlimeHutch.isFull)),
+            postfix: new HarmonyMethod(typeof(ModEntry), nameof(SlimeHutch_isFull_postfix))
+        );
+
+        harmony.Patch(
+            AccessTools.Method(typeof(SlimeHutch), nameof(SlimeHutch.DayUpdate)),
+            prefix: new HarmonyMethod(typeof(ModEntry), nameof(SlimeHutch_DayUpdate_Prefix))
+        );
+
+        // Delay GMCM setup until all mods are initialized
+        helper.Events.GameLoop.GameLaunched += this.OnGameLaunched;
+    }
+
+    private void OnGameLaunched(object sender, EventArgs e)
+    {
+        this.SetupGMCM();
+    }
+
+    private void SetupGMCM()
```

Comment:
Could you split the Generic Mod Config Menu integration into a separate PR? That way it can be reviewed/tweaked/merged separately while we discuss the slime ball changes.

Reply:
Done via PR #100.